docs/android-faqs.md

```diff
@@ -0,0 +1,119 @@
+
+# Rust + Android FAQs
```

Comment:
This used to be in a "how do I expose rust code to kotlin?" howto, but most of the good stuff was actually FAQs, so I removed the no-longer-recommented palaver about manually writing bindings and turned it into an actual FAQ.